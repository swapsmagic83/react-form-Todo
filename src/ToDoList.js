import React, { useState } from "react";
import {v4 as uuid } from "uuid"
import ToDo from "./ToDo";
import NewToDoForm from "./NewToDoForm";


const ToDoList =() =>{
    const initial_state =[]

    const [tasks,setTasks]= useState(initial_state)
    const addTask = (newTask) =>{
        setTasks(tasks =>[...tasks,{id:uuid(),...newTask}])
    }
    const removeTask = (id) =>{
        setTasks(tasks =>tasks.filter(task=>task.id !== id))
    }

    return (
        <>
        <h2>To Do List:</h2>
        <div><NewToDoForm addTask={addTask} /></div>
        
        <div>
            {tasks.map(task => 
            <ToDo 
            text={task.text} 
            id={task.id} 
            key={task.id}
            removeTask={removeTask}/>)}
        </div>
        </>
    )
}

export default ToDoList