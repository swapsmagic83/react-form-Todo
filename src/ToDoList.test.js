import React from "react";
import { render, fireEvent } from "@testing-library/react";
import ToDoList from "./ToDoList";

function addTask(toDoList,text="Laundry"){
    const input= toDoList.getByLabelText("Task")
    const btn =toDoList.queryByText("Add")
    fireEvent.change(input,{target: {value:'Laundry'}})
    fireEvent.click(btn)
}
it("renders without crashing", function(){
    render(<ToDoList />)
})
it("matches snapshot",function(){
    const { asFragment } = render(<ToDoList />)
    expect(asFragment()).toMatchSnapshot()
})

it("should add new item",function(){
    const toDoList= render(<ToDoList />)
    expect(toDoList.queryByText("Remove")).not.toBeInTheDocument();
    addTask(toDoList)
    const removeButton = toDoList.getByText("Remove");
    expect(toDoList.queryByText('Laundry')).toBeInTheDocument()
    expect(removeButton).toBeInTheDocument();
    
})
it("should remove task",function(){
    const toDoList= render(<ToDoList />)
    addTask(toDoList)
    const removeButton = toDoList.getByText("Remove");
    fireEvent.click(removeButton)
    expect(removeButton).not.toBeInTheDocument()
})
