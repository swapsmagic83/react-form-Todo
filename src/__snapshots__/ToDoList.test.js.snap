// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <h2>
    To Do List:
  </h2>
  <div>
    <form>
      <label
        for="text"
      >
        Task
      </label>
      <input
        id="text"
        name="text"
        placeholder="Enter task here"
        type="text"
        value=""
      />
      <button>
        Add
      </button>
    </form>
  </div>
  <div />
</DocumentFragment>
`;
